araproc/analysis/standard_reco.py

```diff
@@ -924,3 +926,346 @@ def get_avg_corr_snr(self, wave_packet, excluded_channels = []):
         avg_hpol_corr_snr = np.mean(avg_hpol_corr_snr)
 
         return avg_vpol_corr_snr, avg_hpol_corr_snr  
+
+    def __get_arrival_delays_reco(
+        self, data, reco_results, channels_to_csw, reference_ch, 
+        which_distance, solution
+    ):
+        """
+        Extract the arrival time of the signals from the reconstruction then 
+        calculate the time delays between each channel and the reference channel
+
+        Parameters
+        ----------
+        data : AnalysisDataset
+        reco_results : dict
+            Reco results already with the specific reconstruction reqeusted (so
+            the keys of this object include 'theta' and 'phi')
+        channels_to_csw : list
+            A list of channels IDs to calculate arrival delays for
+        reference_ch : int
+            Reference channel for use calculating arrival delays
+        which_distance : str
+            The distance of the reconstruction for analysis, following the 
+            convention of other functions in this class. 
+        solution : int
+            0 for direct ray solution. 1 for reflected/refracted ray solution.
+
+        Returns
+        -------
+        arrival_delays : dict
+            Dictionary where keys are channel numbers and values are arrival 
+            delays calculated based on expected arrival times from the 
+            reconstructed event vertex.
+        """
+
+        # Get all the arrival times calculated based on the reconstructed
+        #   event vertex and the time tables
+        arrival_times = {ch: 0.0 for ch in channels_to_csw }
+        for c, ch_ID in enumerate(channels_to_csw):
+            arrival_times[ch_ID] = self.lookup_arrival_time(
+                ch_ID, reco_results['theta'], reco_results['phi'], 
+                which_distance=which_distance, solution=solution
+            )
+        
+        # Calculate time difference between each channel's arrival time
+        #   and the reference channel's arrival time
+        reference_arrival_time = arrival_times[reference_ch]
+        arrival_delays = {ch: 0.0 for ch in channels_to_csw }
+        for ch_ID in arrival_delays.keys():
+            arrival_delays[ch_ID] = arrival_times[ch_ID] - reference_arrival_time
+
+        return arrival_delays
+
+    def __get_arrival_delays_AraRoot_xcorr(
+        self, wavepacket, pol, channels_to_csw, reference_ch, reco_delays, 
+        trig_type, which_distance, zoom_window=40
+    ):
+        """
+        Determine the arrival delays from each channel by finding the time of
+        max correlation between each channel and the reference channel in a 
+        `zoom_window` nanosecond window around the expected arrival delay of a 
+        signal given the results from reconstructing the event.
+
+        Parameters
+        ----------
+        wavepacket : dict
+            The wavepacket where keys are not yet channel numbers.
```

Comment:
What does this mean? "where keys are not yet channel numbers"

Reply:
Like the wavepacket has keys of `event` and `waveforms` and the value corresponding to the key of `waveforms` is a dictionary where keys are channel numbers. I wasn't sure how to phrase this comment in a way that would be less likely to need updating if we change the way the wavepackets work in the future. 

Maybe I can say "Wavepacket where keys include waveforms and the event number" if that's better? Or if we're going to be consistent in differentiating 'waveforms' and 'wavepackest' then maybe I'm overthinking this

Reply:
I changed it to this: "AraProc wavepacket (keys include waveforms and the event number)"